### DISCUSSION TITLE
Set maximal/default Expiry of AccessKeys

### DISCUSSION THREAD

**Discussion by julir1** [minio/minio#20386] [Q&A]
Hello together

I have installed a Minio ObjectStorage Server on an Ubuntu at home.
This is now connected to OpenID via Microsoft Entra. The login based on the AAD groups and Minio policies works perfectly. So far everything is fine. 

Now I would like to provide a little more security for the access keys:

- I would like to ensure globally as admin that an access key can be valid for a maximum of one year (Currently, a user can generate an access key after logging in, which exists indefinitely. For example, do not set an expiration date at all)
- Ensure that the secret key of the access key is always generated and cannot be changed.

![image](https://github.com/user-attachments/assets/b2d1c621-2fbd-47c4-a0e5-0b394d69f56a)

Is this possible? if so, where can I set this?

Thanks!

**Comment by wrouesnel** [comment]
+1 for this. Using LDAP login users can issue their own access keys, but nothing stops them making an infinite lifetime access key that means and then treating it however they want. Ideally I'd like to limit those keys to 12 hours or so (i.e. slightly more then a work day) to keep the attack surface under control.

**Comment by prakashsvmx** [comment]
There is no option/feature to set  expiry of User's Access Key/Service Accounts globally
